### DISCUSSION TITLE
a Kernel for del(V) * del(a function of T)

### DISCUSSION THREAD

**Discussion by bosxered** [idaholab/moose#24245] [Q&A General] [answered]
Dear Moose experts,

Hello.
I want to solve the below heat conduction equation. In order to solve the equation, I have to custom(make) a kernel for myself or use a kernel that has been already made and used.

![image](https://user-images.githubusercontent.com/79071957/235675920-03cee2b8-19f8-430d-97ab-73140b7a3fd0.png)

So, there are two variables V(voltage), and T(temperature).

Maybe, I can use the HeatConductionTimeDerivative, HeatConduction, JouleHeatingSource, and MatDiffusion kernels to solve the equation.
The problem is the -sigma*del(V)*del(pi) term.
**Is there any kernel that the Moose provides? Should I make a new one for myself?**

In addition, if I have wrong thinking or derivative formula, please feel free to tell me.

For your information, another equation is -sigma*del^2(V)=0; a current continuity equation that has no time derivative.

**Comment by GiudGiud** [comment, marked as answer]
Hello

This last term does not look familiar so I m going to venture that we do not have a premade version.

Making kernels should not be difficult if you followed the tutorial. 
There's also plenty of examples. 

An option btw, is to use an integration by parts on that term that will avoid requiring to take the derivative of that term. 
Instead you would have J Pi grad_test for the residual

Guillaume

**Reply by bosxered** [reply]
Thank you so much!
